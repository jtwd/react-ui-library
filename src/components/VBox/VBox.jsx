import React from 'react'
import { node, oneOf } from 'prop-types'
import styled from 'styled-components'

import { alignToFlex, packToFlex } from "../_theme/mixins/flexbox"

const StyledDiv = styled.div`
  display: flex;
  flex-direction: column;
  
  ${props => props.align && `
    align-items: ${alignToFlex[props.align]};
  `}
  
  ${props => props.pack && `
    justify-content: ${packToFlex[props.pack]};
  `}
`

/** Flexbox layout helper for vertical alignment */
function VBox({ children, align, pack, ...props}) {
  return (
    <StyledDiv align={align} pack={pack} {...props}>{children}</StyledDiv>
  )
}

VBox.propTypes = {
  /** stretch, top, bottom, center */
  align: oneOf(['stretch', 'top', 'bottom', 'center']),
  /** start, center, ends, between, around */
  pack: oneOf(['start', 'center', 'end', 'between', 'around', null]),
  children: node.isRequired
}

VBox.defaultProps = {
  align: 'stretch',
  pack: null
}

export default VBox
