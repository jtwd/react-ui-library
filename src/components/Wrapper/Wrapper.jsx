import React from 'react'
import { node, bool } from 'prop-types'
import styled from 'styled-components'
import { md } from '../../theme/spacers'
import { textMaxWidth, shortMaxWidth } from '../../theme/units'

const StyledDiv = styled.div`
  padding: ${md};
  
  ${props => props.text && `
    max-width: ${textMaxWidth};
  `}
  
  ${props => props.short && `
    max-width: ${shortMaxWidth};
  `}
  
  ${props => props.centered && `
    margin-left: auto;
    margin-right: auto;
  `}
`

function Wrapper(props) {
  const { text, centered, short } = props
  return (
    <StyledDiv text={text} centered={centered} short={short}>
      {props.children}
    </StyledDiv>
  )
}

Wrapper.propTypes = {
  /** Sized appropriately for text */
  text: bool,
  /** Short Sized  */
  short: bool,
  /** Centered wrapper */
  centered: bool,
  /** Contents of wrapper */
  children: node
}

Wrapper.defaultProps = {
  text: false,
  short: false,
  centered: false,
  children: null
}

export default Wrapper
