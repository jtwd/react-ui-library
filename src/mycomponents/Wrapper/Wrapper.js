import styled from 'styled-components'
import { md } from '../../theme/spacers'

const Wrapper = styled.div`
  padding: ${md};
`

export default Wrapper
