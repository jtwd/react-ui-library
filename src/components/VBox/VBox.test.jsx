import React from 'react'
import {shallow} from 'enzyme'

import VBox from './VBox'
import Wrapper from '../Wrapper'

describe('VBox component', () => {
  it('renders correctly by default (stretched)', () => {
    const component = shallow(
      <VBox>
        <Wrapper flex={1}>1</Wrapper>
        <Wrapper flex={1}>2</Wrapper>
      </VBox>
    )
    expect(component).toMatchSnapshot()
  })
  it('renders top aligned', () => {
    const component = shallow(
      <VBox align="top">
        <Wrapper flex={1}>1</Wrapper>
        <Wrapper flex={1}>2</Wrapper>
      </VBox>
    )
    expect(component).toMatchSnapshot()
  })
  it('renders bottom aligned', () => {
    const component = shallow(
      <VBox align="bottom">
        <Wrapper flex={1}>1</Wrapper>
        <Wrapper flex={1}>2</Wrapper>
      </VBox>
    )
    expect(component).toMatchSnapshot()
  })
  it('renders packed to the start', () => {
    const component = shallow(
      <VBox pack="start">
        <Wrapper>1</Wrapper>
        <Wrapper>2</Wrapper>
      </VBox>
    )
    expect(component).toMatchSnapshot()
  })
  it('renders packed in the center', () => {
    const component = shallow(
      <VBox pack="center">
        <Wrapper>1</Wrapper>
        <Wrapper>2</Wrapper>
      </VBox>
    )
    expect(component).toMatchSnapshot()
  })
  it('renders packed to the end', () => {
    const component = shallow(
      <VBox pack="end">
        <Wrapper>1</Wrapper>
        <Wrapper>2</Wrapper>
      </VBox>
    )
    expect(component).toMatchSnapshot()
  })
  it('renders packed with space between', () => {
    const component = shallow(
      <VBox pack="between">
        <Wrapper>1</Wrapper>
        <Wrapper>2</Wrapper>
      </VBox>
    )
    expect(component).toMatchSnapshot()
  })
  it('renders packed with space around', () => {
    const component = shallow(
      <VBox pack="around">
        <Wrapper>1</Wrapper>
        <Wrapper>2</Wrapper>
      </VBox>
    )
    expect(component).toMatchSnapshot()
  })
})
