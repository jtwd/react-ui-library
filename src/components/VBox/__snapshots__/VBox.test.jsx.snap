// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VBox component renders bottom aligned 1`] = `
<styled.div
  align="bottom"
  pack={null}
>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={1}
    middle={false}
    short={false}
    text={false}
  >
    1
  </Wrapper>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={1}
    middle={false}
    short={false}
    text={false}
  >
    2
  </Wrapper>
</styled.div>
`;

exports[`VBox component renders correctly by default (stretched) 1`] = `
<styled.div
  align="stretch"
  pack={null}
>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={1}
    middle={false}
    short={false}
    text={false}
  >
    1
  </Wrapper>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={1}
    middle={false}
    short={false}
    text={false}
  >
    2
  </Wrapper>
</styled.div>
`;

exports[`VBox component renders packed in the center 1`] = `
<styled.div
  align="stretch"
  pack="center"
>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    1
  </Wrapper>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    2
  </Wrapper>
</styled.div>
`;

exports[`VBox component renders packed to the end 1`] = `
<styled.div
  align="stretch"
  pack="end"
>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    1
  </Wrapper>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    2
  </Wrapper>
</styled.div>
`;

exports[`VBox component renders packed to the start 1`] = `
<styled.div
  align="stretch"
  pack="start"
>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    1
  </Wrapper>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    2
  </Wrapper>
</styled.div>
`;

exports[`VBox component renders packed with space around 1`] = `
<styled.div
  align="stretch"
  pack="around"
>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    1
  </Wrapper>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    2
  </Wrapper>
</styled.div>
`;

exports[`VBox component renders packed with space between 1`] = `
<styled.div
  align="stretch"
  pack="between"
>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    1
  </Wrapper>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={null}
    middle={false}
    short={false}
    text={false}
  >
    2
  </Wrapper>
</styled.div>
`;

exports[`VBox component renders top aligned 1`] = `
<styled.div
  align="top"
  pack={null}
>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={1}
    middle={false}
    short={false}
    text={false}
  >
    1
  </Wrapper>
  <Wrapper
    app={false}
    centered={false}
    collapse={false}
    flex={1}
    middle={false}
    short={false}
    text={false}
  >
    2
  </Wrapper>
</styled.div>
`;
